code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/dao/impl/ServiceExposeMapDaoImpl.java

```diff
@@ -194,4 +196,14 @@ public ServiceExposeMap getServiceIpExposeMap(Service service, String ipAddress)
                 SERVICE_EXPOSE_MAP.REMOVED, null);
     }
 
+    @Override
+    public List<? extends Service> getGlobalServices(long accountId) {
+        return create()
+                .select(SERVICE.fields())
+                .from(SERVICE)
+                .where(SERVICE.ACCOUNT_ID.eq(accountId))
+                .and(SERVICE.REMOVED.isNull())
+                .and(SERVICE.STATE.in(CommonStatesConstants.ACTIVE, CommonStatesConstants.ACTIVATING))
```

Comment:
Sorry, should have included it earlier - we should also include updating-active state here.